sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -49,7 +49,7 @@ case class Mode(
   // Returns null for empty inputs
   override def nullable: Boolean = true
 
-  override val dataType: DataType = child.dataType
+  override def dataType: DataType = child.dataType
```

Comment:
this change is needed, otherwise:
```
scala> df.groupBy("a").agg(mode(col("b")))
org.apache.spark.sql.catalyst.analysis.UnresolvedException: Invalid call to dataType on unresolved object
  at org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute.dataType(unresolved.scala:137)
  at org.apache.spark.sql.catalyst.expressions.aggregate.Mode.<init>(Mode.scala:52)
  at org.apache.spark.sql.functions$.mode(functions.scala:679)
  ... 49 elided

```